qiita_files/demux.py

```diff
@@ -455,6 +461,60 @@ def to_per_sample_ascii(demux, samples=None):
         yield samp, to_ascii(demux, samples=[samp])
 
 
+def _to_file(demux_fp, sample, fp, formatter):
+    with open_file(demux_fp, 'r+') as demux:
+        with open(fp, 'wb') as out:
+            for rec in _to_ascii(demux, [sample], formatter):
+                out.write(rec)
+
+
+def to_per_sample_files(demux_fp, samples=None, out_dir='./', n_jobs=1,
+                        out_format='fastq'):
+    """Writes per sample files
+
+    Parameters
+    ----------
+    demux_fp : str
+        The demux file path
+    samples : list of str, optional
+        Samples to pull out. If None, then all samples will be examined.
+        Defaults to None.
+    out_dir : str, optional
+        Path to output directory to store the per sample fasta.
+        Defaults to current directory
+    n_jobs : int, optional
+        Number of jobs to run in parallel. Defaults to 1
+    out_format : {'fastq', 'fasta'}
+        The format in which the output files should be written.
+    """
+    if out_format == 'fastq':
+        formatter = format_fastq_record
+        file_name_fmt = "%s.fq"
```

Comment:
I saw that in the tests 'fq' was used, that's why I used that, but no strong feelings so I can change.